src/main/java/org/commcare/suite/model/StackFrameStep.java

```diff
@@ -234,4 +236,13 @@ public void setType(String elementType) {
     public String getElementType() {
         return elementType;
     }
+
+    // find and replace any arguments $argument with their value in {@code arguments}
+    public void substituteArgument(HashMap<String, String> arguments) {
+        if (value.indexOf("$") != -1) {
```

Comment:
I was imagining that we would provide arguments through the normal XPath engine. You can set variables in the Evaluation Context, and they will be evaluated as normal.

Munging the string in place can come with pretty complex issues, especially if the user inputs are complex